[{"items": [{"tags": ["python-3.x", "python-2.7", "tensorflow"], "owner": {"account_id": 13791835, "reputation": 147, "user_id": 9954063, "user_type": "registered", "profile_image": "https://graph.facebook.com/2236761996363870/picture?type=large", "display_name": "Sai Raghava", "link": "https://stackoverflow.com/users/9954063/sai-raghava"}, "is_answered": false, "view_count": 41, "answer_count": 0, "score": 1, "last_activity_date": 1567527340, "creation_date": 1567527340, "question_id": 57775393, "content_license": "CC BY-SA 4.0", "link": "https://stackoverflow.com/questions/57775393/how-to-solve-failedpreconditionerror", "title": "How to solve FailedPreconditionError?", "body": "<p>I have a small 3 layer network where I have created variables for weights and biases using initer, and I have used <code>tf.variables_initializer</code>to initialize variables as suggested by someone. Unfortunately despite initializin them, there still apears a Failed PreCondition Error. I am now out of ideas about how to solve it or what exactly is causing it.</p>\n\n<pre><code>with tf.compat.v1.variable_scope(\"weight_in\", reuse=tf.compat.v1.AUTO_REUSE):\n            weight_i = tf.compat.v1.get_variable(\"weight_input\", dtype=tf.float64, shape=[self.neurons, 1], initializer=self.weight_initer)\n        with tf.compat.v1.variable_scope(\"weight_hid\", reuse=tf.compat.v1.AUTO_REUSE):   \n            weight_h = tf.compat.v1.get_variable(\"weight_hidden\", dtype=tf.float64, shape=[self.neurons, 1], initializer=self.weight_initer1)\n        with tf.compat.v1.variable_scope(\"weight_out\", reuse=tf.compat.v1.AUTO_REUSE): \n            weight_o = tf.compat.v1.get_variable(\"weight_input\", dtype=tf.float64, shape=[4, 2], initializer=self.weight_initer2)\n        with tf.compat.v1.variable_scope(\"bias_in\", reuse=tf.compat.v1.AUTO_REUSE): \n            bias_i = tf.compat.v1.get_variable(\"bias_input\", dtype=tf.float64, shape=[self.neurons, 1], initializer=self.bias_initer)\n        with tf.compat.v1.variable_scope(\"bias_hid\", reuse=tf.compat.v1.AUTO_REUSE): \n            bias_h = tf.compat.v1.get_variable(\"bias_hidden\", dtype=tf.float64, shape=[self.neurons, 1], initializer=self.bias_initer1)\n        with tf.compat.v1.variable_scope(\"bias_out\", reuse=tf.compat.v1.AUTO_REUSE): \n            bias_o = tf.compat.v1.get_variable(\"bias_out\", dtype=tf.float64, shape=[1, 4], initializer=self.bias_initer2)\n       uninitialized_variables = []\n        sess = tf.Session()\n        print(\"printing the names of all variables\")\n        print(tf.all_variables())\n        for var in tf.all_variables():\n            try:\n                sess.run(var)\n            except tf.errors.FailedPreconditionError:\n                uninitialized_variables.append(var)\n        init = tf.variables_initializer(uninitialized_variables)\n        sess.run(init)\n</code></pre>\n\n<p>The error is </p>\n\n<pre><code>FailedPreconditionError: Attewmpting to use initialized value weight_in/weight_input\n</code></pre>\n\n<p>sometimes the same error will be pointing another bias variable. I am wondering whether anyone can shed some light on how to properly create a variable and initialize it. Thank you.</p>\n"}], "has_more": false, "quota_max": 300, "quota_remaining": 97}]